### DISCUSSION TITLE
Mealie not requesting groups scope from authelia

### DISCUSSION THREAD

**Discussion by EinToni** [mealie-recipes/mealie#3809] [Q&A] [answered]
Hey everybody,
I currently try to setup mealie to accept authelia for authentification. It is already working fine with the exception that the groups are somehow not read. If I remove `OIDC_USER_GROUP` and `OIDC_ADMIN_GROUP` the user gets created and logged in and everything is fine, but I would like to have an admin user and therefore want to use these tags.
When requesting the login via authelia, mealie is somehow not requesting the groups scope:

![Authelia Login Screen](https://github.com/mealie-recipes/mealie/assets/51962051/8c10b2e2-45b8-4bcf-9dbf-b21eb9e73ce9)

The log of mealie additionally shows that the groups are empty:
```
DEBUG    2024-06-27T20:58:06 - [OIDC] User does not have the required group. Found: [] - Required: mealie-users
ERROR    2024-06-27T20:58:06 - Incorrect username or password from 192.168.0.46
ERROR    2024-06-27T20:58:06 - Incorrect username or password from 192.168.0.46
INFO     2024-06-27T20:58:06 - [192.168.0.46:0] 401 Unauthorized "POST /api/auth/token HTTP/1.1"
DEBUG    2024-06-27T20:58:06 - [OIDC] User does not have the required group. Found: [] - Required: mealie-users
ERROR    2024-06-27T20:58:06 - Incorrect username or password from 192.168.0.46
ERROR    2024-06-27T20:58:06 - Incorrect username or password from 192.168.0.46
INFO     2024-06-27T20:58:06 - [192.168.0.46:0] 401 Unauthorized "POST /api/auth/token HTTP/1.1"
INFO     2024-06-27T20:58:06 - [192.168.0.46:0] 307 Temporary Redirect "GET /login?direct=1 HTTP/1.1"
INFO     2024-06-27T20:58:06 - [192.168.0.46:0] 200 OK "GET /api/app/about HTTP/1.1"
INFO     2024-06-27T20:58:06 - [192.168.0.46:0] 200 OK "GET /api/app/about HTTP/1.1"
INFO     2024-06-27T20:58:06 - [192.168.0.46:0] 200 OK "GET /api/app/about/startup-info HTTP/1.1"
INFO     2024-06-27T20:58:06 - [192.168.0.46:0] 200 OK "GET /api/app/about HTTP/1.1"
INFO     2024-06-27T20:58:06 - [192.168.0.46:0] 200 OK "GET /api/app/about HTTP/1.1"
```

Even tho I have authelia set up to use a file as user-database. Here I have configured the groups, so they should work:
```
users:
  testuser:
    disabled: false
    displayname: "<xxx>"
    password: "<xxx>"
    email: <xxx>
    groups:
      - mealie-users
      - mealie-admins
```

And I have configured the client in authelia to allow the necessary scopes:
```
      -
        client_id: 'mealie'
        client_name: 'Mealie'
        authorization_policy: one_factor
        require_pkce: true
        pkce_challenge_method: 'S256'
        public: true
        grant_types:
          - 'authorization_code'
        scopes:
          - openid
          - profile
          - groups
          - email
        redirect_uris:
          - https://mealie.<domain>/login
        userinfo_signed_response_alg: 'none'
        token_endpoint_auth_method: 'none'
```

I am using mealie v1.9.0 build [d96c36333b9cb9461c5dee96ae28b60d912b38fd](https://github.com/mealie-recipes/mealie/commit/d96c36333b9cb9461c5dee96ae28b60d912b38fd)
But I also tried v1.8.0 and v1.7.0 so I strongly assume it's a configuration problem on my side. I tried to get that to work the last hour but just have no idea why these groups are empty. Hopefully someone of you can provide additional support and maybe share their config or sees an obvious error on my side.
(Addition info: I am using ngingx proxy manager and set that up to use authelia according to [authelias page]( https://www.authelia.com/integration/proxies/nginx-proxy-manager/#protected-application))

**Comment by cmintey** [comment, marked as answer]
Make sure you set `OIDC_GROUPS_CLAIM=groups` (this is the default), and you have a user or admin group defined in mealie as well.

If you have both of that already, then try resetting your browser cache or use a private browser. There is an API which is called to determine if the browser will request the groups scope and that call can be cached

**Reply by EinToni** [reply]
The `OIDC_GROUPS_CLAIM` was already set but what helped was.... clearing the browser cache. I can't believe it but something as simple as that solved it. Thanks!
